scripts/components/Sidebar.js

```diff
@@ -1,43 +1,90 @@
+              <Link to={`/buckets/${bucket.id}/collections/${id}/edit`}
+                className="collections-menu-entry-edit"
+                title="Edit collection properties">
+                <i className="glyphicon glyphicon-cog"/>
+              </Link>
+            </div>
+          );
+        })
+      }
+      <Link className="list-group-item"
+        to={`/buckets/${bucket.id}/create-collection`}>
+        <i className="glyphicon glyphicon-plus"/>
+        Create collection
+      </Link>
+    </div>
+  );
+}
+
+function BucketsMenu(props) {
+  const {buckets, userBucket, bid, cid} = props;
+  return (
+    <div>{
+      buckets.map((bucket, i) => {
+        const {id, collections} = bucket;
+        const current = bid === id;
+        return (
+          <div key={i} className="panel panel-default bucket-menu">
+            <div className="panel-heading">
+              <i className={`glyphicon glyphicon-folder-${current ? "open" : "close"}`} />
+              <strong>{id === userBucket ? "default" : id}</strong> bucket
+            </div>
+            <BucketCollectionsMenu
+              bucket={bucket}
+              collections={collections}
+              bid={bid}
+              cid={cid} />
+          </div>
+        );
+      })
+    }</div>
+  );
+}
+
+export default class Sidebar extends Component {
   render() {
-    const {collections, params} = this.props;
+    const {session, params, location} = this.props;
+    const {bid, cid} = params;
+
+    function activeIfPathname(pathname) {
+      const active = location.pathname === pathname ? "active" : "";
+      return `list-group-item ${active}`;
+    }
+
+    const {buckets} = session;
     return (
       <div>
         <div className="panel panel-default">
           <div className="list-group">
-            <Link to="/" className={this.activeIfPathname("/")}>Home</Link>
-            <Link to="/settings"
-              className={this.activeIfPathname("/settings")}>Settings</Link>
+            <Link to="/" className={activeIfPathname("/")}>Home</Link>
           </div>
         </div>
-
-        <div className="panel panel-default">
-        <div className="panel-heading">
-          Collections
-        </div>
-          <div className="list-group">{
-            Object.keys(collections).map((name, i) => {
-              const classes = [
-                "list-group-item",
-                params.name === name ? "active" : "",
-              ].join(" ");
-              const label = collections[name].name || name;
-              return (
-                <Link key={i} to={`/collections/${name}`} className={classes}>
-                  {label + (!collections[name].synced ? "*" : "")}
-                </Link>
-              );
-            })
-          }</div>
-        </div>
+        {session.authenticated ?
+          <BucketsMenu
+            buckets={buckets}
+            userBucket={session.serverInfo.user.bucket}
```

Comment:
This will be `undefined` if no default plugin. From what I read below, it won't pose any problem.